compiler/semmagic.nim

```diff
@@ -575,7 +575,8 @@ proc magicsAfterOverloadResolution(c: PContext, n: PNode,
     result = n
   of mPrivateAccess:
     var t = n[1].typ[0]
-    if t.kind in {tyRef, tyPtr}: t = t[0]
+    if t.kind in {tyRef, tyPtr, tyGenericInst}: t = t[0]
```

Comment:
Still not correct you need something like:

```nim

var indirections = 0
while true:
  case t.kind 
  of tyRef, tyPtr:
    inc indirections 
    t = t[0]
  of tyGenericInst:
    t = t[0] # really? not  t.lastSon ??
  else:
    break

if indirections <= 1:
  ...
```

If the purpose was to skip a single pointer, that is.

Reply:
thanks, should be correct with latest commit; your version still wasn't quite correct but I've corrected it (also `t.lastSon` wouldn't work for the tyGenericInst case as can be seen by looking at `astalgo.debug`)

the new tests i added reflect that